.github/workflows/katello_react_tests.yml

```diff
@@ -0,0 +1,37 @@
+name: (non-blocking) Katello React
+
+on:
+  pull_request:
+    branches: [develop]
+
+defaults:
+  run:
+    working-directory: ./katello
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          path: ${{ github.workspace }}/katello
```

Comment:
You are not specifying a repository here, are you sure it'll checkout Katello? I'd assume it'll checkout the repository it runs on by default.
Take a look at for example:
https://github.com/theforeman/foreman_statistics/blob/a2d90e6dee9350b57fbaa5baf8f9ceec1944f922/.github/workflows/ci.yml#L40